### PR TITLE
Automate search index builds

### PR DESCRIPTION
**Pull request by sivakumar-kailasam**
This should address https://github.com/ember-learn/guides-source/issues/487. It's been a while since we've had that issue open. I suggest we begin with these changes as a starting point & improve on it to address any other concerns we may have about it.

Once this PR is merged, we can trigger rebuilds whenever needed by creating tags.

### CODE REVIEW

**Comment by jenweber** [comment]
Here are the things this PR would need to do:

- Whenever there is a new release, run the search indexer
- Only update indexes for the latest version of Ember (i.e. don't reindex all versions)
- Update maintainers.md with any relevant instructions, or remove instructions that aren't needed (this could be a followup PR)

**Comment by sivakumar-kailasam** [comment]
@jenweber the last couple of commits should address your PR.

File: MAINTAINERS.md [hunk 1]
```diff
@@ -34,35 +34,6 @@ It is required that all maintainers use 2FA (two factor authentication). These a
...
-  - rm -rf v*
-2. Open versions.yml and delete everything in `allVersions` apart from the latest version (that has just been released) 
-3. Make sure `config/credentials.json` is in place as described above
-  - you may have accidentally deleted this since last time as it is supposed to be ignored by git
-4. Open `config/deploy.js` and delete the `versionsToIgnore` line in the `prember-algolia` config
-5. run `ember deploy production`
-6. This should now be done, you can fix your local repo by running `git reset --hard HEAD`
-7. Before you walk away, you should check the guides app in production and see if you can search for something on the latest version
-8. Party some more 🎉and let the rest of the team know that the updates have been made.  
+1. Create a git tag which would trigger a github action that'll take care of updating the indexes.
```
Comment (jenweber) [inline MAINTAINERS.md]: Does any tag trigger the release? Maybe this should say:

```suggestion
1. Using the GitHub Releases page, create a git tag for the release version, i.e. `v3.16.0`. Tagging a release using any tag name triggers a github action that'll take care of updating the indexes.
```

File: prerelease-setup.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  guidesVersionDoc.allVersions = [guidesVersionDoc.allVersions.pop()];
+  fs.writeFileSync(versionsFile, YAML.stringify(guidesVersionDoc));
+
+  const client = algoliaSearch(
+    process.env.ALGOLIA_APPLICATION,
+    process.env.ALGOLIA_KEY
+  );
+
+  const guidesSearchIndex = client.initIndex(process.env.ALGOLIA_INDEX);
+  await guidesSearchIndex.clearIndex();
```
Comment (jenweber) [inline prerelease-setup.js]: What does clearIndex do here?

Comment (sivakumar-kailasam) [reply prerelease-setup.js]: it clears the algolia index. @mansona mentioned something about clearing indexes before publishing new indexes on discord the other day. Maybe he can confirm if this is required

**Comment by sivakumar-kailasam** [comment]
Thanks @jenweber, I'll wait for @mansona to confirm the behavior about clearing indexes before shipping this.

**Comment by sivakumar-kailasam** [comment]
@mansona can you please have a look at this PR & let me know if we require any changes?

**Comment by amyrlam** [comment]
Do we want to try this? @mansona what do you think?

**Comment by jenweber** [comment]
This PR is superseded by/included in work being done by @locks to automate releases. See the `tool-new-release` repo. The new work addresses some of the instability we've had in the guides search lately.

**Pull request closed without merging**